src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Acquisition/MissingSetupComponentRegistrationService.cs

```diff
@@ -37,6 +45,37 @@ internal class MissingSetupComponentRegistrationService : IMissingSetupComponent
         private IVsSolution? _vsSolution;
         private bool? _isVSFromPreviewChannel;
 
+        private readonly object _lock = new();
+        private static HashSet<string>? _netCoreRegistryKeyValues = null;
+
+        private HashSet<string> NetCoreRegistryKeyValues
+        {
+            get
+            {
+                if (_netCoreRegistryKeyValues is null)
+                {
+                    lock (_lock)
+                    {
+                        if (_netCoreRegistryKeyValues is null)
+                        {
+                            var regkey = RegistryKey.OpenBaseKey(RegistryHive.LocalMachine, RegistryView.Registry64);
```

Comment:
Again... only works on 64-bit Windows?

Reply:
What I meant that is there is no 64-bit registry on a 32-bit Windows (or at least that's how it used to be). There are explicitly 2 different registries for 32-bit versus 64-bit.
Read this: https://serverfault.com/questions/7973/is-there-a-distinction-between-64-bit-and-32-bit-regedit

Reply:
VS has its own registry hive, so I don't think this is a concern.

Reply:
This is looking at a machine-wide part of the registry, which is the point. If this information were available in the VS-instance-specific part of the registry, we probably wouldn't need to look at the registry at all.

But as VS is 64-bit (and it explicitly requires a 64-bit CPU and 64-bit Windows) I don't think this is much of a concern.